File: Đồ Án OOP/Đồ Án OOP/KyNang.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Đồ_Án_OOP
{
    class KyNang : IKyNang
    {
        private string sTenKyNang;

        private string sLoaiKyNang;

        private string sMieuTa;

        public string STenKyNang { get => sTenKyNang; set => sTenKyNang = value; }
        public string SLoaiKyNang { get => sLoaiKyNang; set => sLoaiKyNang = value; }
        public string SMieuTa { get => sMieuTa; set => sMieuTa = value; }

        public KyNang()
        {
            this.sTenKyNang = "";
            this.SLoaiKyNang = "";
            this.sMieuTa = "";
        }

        public KyNang(string tenkn, string loaikn, string mieuta)
        {
            this.STenKyNang = tenkn;
            this.SLoaiKyNang = loaikn;
            this.SMieuTa = mieuta;
        }

        public void Nhap()
        {
            Console.WriteLine("Nhap ten ky nang: ");
            this.STenKyNang = Console.ReadLine();
            int a;
            do
            {
                Console.WriteLine("Nhap loai ky nang: \n1: CHU DONG.\n2: BI DONG.\n");
                a = int.Parse(Console.ReadLine());
            } while (a != 1 && a != 2);
            if (a == 1)
                this.SLoaiKyNang = "Chu dong.";
            else
                this.SLoaiKyNang = "Bi dong.";

            Console.WriteLine("Mieu ta: ");
            this.SMieuTa = Console.ReadLine();
        }

        public void Nhap(string tenkn, int loaikn, string mieuta)
        {
            this.sTenKyNang = tenkn;

            if (loaikn == 2)
                this.SLoaiKyNang = "Bi dong";
            else
                this.SLoaiKyNang = "Chu dong";

            this.sMieuTa = mieuta;
        }

        public void DocFile()
        {
            string line = "";
            StreamReader file = new StreamReader("KyNang.txt");
            while ((line = file.ReadLine()) != null)
            {
                this.sTenKyNang = line;
                this.sLoaiKyNang = file.ReadLine();
                this.sMieuTa = file.ReadLine();
            }
            file.Close();
        }

        public void Xuat()
        {
            Console.WriteLine("- Ten ky nang:" + this.sTenKyNang);
            Console.WriteLine("- Loai ky nang: " + this.SLoaiKyNang);
            Console.WriteLine("- Mieu ta: " + this.sMieuTa);
        }
    }
}

File: C/Đồ_Án_OOP_JAVA/src/KyNang.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Scanner;
import java.io.FileReader;
public abstract class KyNang implements IKyNang
{
	private String sTenKyNang;
	private String sLoaiKyNang;
	private String sMieuTa;
	public String getsTenKyNang() {
		return sTenKyNang;
	}
	public void setsTenKyNang(String sTenKyNang) {
		this.sTenKyNang = sTenKyNang;
	}
	public String getsLoaiKyNag() {
		return sLoaiKyNang;
	}
	public void setsLoaiKyNag(String sLoaiKyNang) {
		this.sLoaiKyNang = sLoaiKyNang;
	}
	public String getsMieuTa() {
		return sMieuTa;
	}
	public void setsMieuTa(String sMieuTa) {
		this.sMieuTa = sMieuTa;
	}
	public KyNang()
	{
		this.sTenKyNang="";
		this.sLoaiKyNang="";
		this.sMieuTa="";
	}
	public KyNang(String tenkn, String loaikn, String mieuta)
	{
		this.sTenKyNang=tenkn;
		this.sLoaiKyNang=loaikn;
		this.sMieuTa=mieuta;
	}
	@SuppressWarnings("resource")
	public void Nhap() throws IOException
	{
		System.out.println("Nhap ten ky nang: ");
		InputStreamReader tenkn=new InputStreamReader(System.in);
		BufferedReader tkn=new BufferedReader(tenkn);
		this.sTenKyNang=tkn.readLine();
		System.out.println("Nhap a: ");
		//BufferedReader in = null;
		int a;		
		Scanner input=new Scanner(System.in);
		do {
			System.out.println("Nhap loai ky nang: \n1. CHU DONG.\n2. BI DONG.\n");
			a=input.nextInt();
		} while (a!=1 && a!=2);
		if (a==1)
			this.sLoaiKyNang="Chu dong.";
		else this.sLoaiKyNang="Bi dong.";
		System.out.println("Mieu ta: ");
		InputStreamReader mta=new InputStreamReader(System.in);
		BufferedReader mt=new BufferedReader(mta);
		this.sMieuTa=mt.readLine();
	}
	public void Nhap(String tenkn, int loaikn, String mieuta)
	{
		this.sTenKyNang=tenkn;
		if(loaikn==2)
			this.sLoaiKyNang="Bi dong.";
		else this.sLoaiKyNang="Chu dong.";
		this.sMieuTa=mieuta;
	}
	public void Xuat()
	{
		System.out.println("- Ten ky nang: "+this.sTenKyNang);
		System.out.println("- Loai ky nang: "+this.sLoaiKyNang);
		System.out.println("- Mieu ta: "+this.sMieuTa);
	}
	public void DocFile()
	{
		//String line="";//
		BufferedReader br=null;
		try {
			br=new BufferedReader(new FileReader("KyNang.txt"));
			System.out.println("Noi dung file: ");
			String line;
			while ((line=br.readLine())!=null)
			{
				System.out.println(line);
				line=br.readLine();
			}
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}
